### ISSUE TITLE
Multi-Tenancy - possible solution

### ISSUE THREAD

**Issue by colinmackinlay** [beyondcode/invoker-community#79]
Relates to #38 (duplicate) and #51 

Posting again as suggested by @sschlein as I have a suggested approach that might be fruitful.

Multitenancy with multiple databases: there is no out-of-the-box way for Invoker to know this and to provide a selection of which tenant database to render. I doubt it would make sense for you to write code for each different MultiTenancy approach but perhaps Invoker could handle this in terms of a method or trait in the models. 

I happen to use https://github.com/stancl/tenancy from @stancl and so my central models all work fine in Invoker as they use the default connection. They do have a method to identify that they should use the central connection but that isn't needed by Invoker:
```
    public function getConnectionName()
    {
        return config('tenancy.database.central_connection');
    }
```
 
There's nothing specific in the Tenant models to identify them so of course Invoker looks for them on the default connection and can't find them there. But they could have a method that tells Invoker to look for a set of tenant databases that could be selectable. If it returned a list of databases for tenants (perhaps with user-friendly names) then Invoker could provide a choice of which tenant database to use when rendering these.

**Comment by mpociot** [comment]
I think having the proposed solution in #68 would be good for your use case as well, wouldn't it?
So you could configure a "pre-load" method, that would modify the config value on a project level.

**Comment by colinmackinlay** [comment]
Yes - it might. Certainly for testing locally where there would probably be a single tenant. 

If using it on a live system though it would be helpful to be able to switch between any of the tenant databases so the solution in #51 for hot-swapping the database would be the luxury solution :)

**Comment by mpociot** [comment]
I just published version 2.1.0, which adds support for custom "preload code" that you can specify per project.
This code gets evaluated before Invoker does it's thing, so you can use it to override configuration values, etc.
